alphagov/paas-release-ci
# [#172429953] Upload remaining secrets to credhub

What
----
Uploads secrets that aren't specific to one or two pipelines to Credhub i.e. those not covered in #136 .

How to review
-------------

1. Create a new build env from paas-bootstrap (you’ll need to add the new entries in Google API console, so that SSO works): `make dev build-concourse bootstrap DEPLOY_ENV=test-ci `

1. Get your dev envs CF admin credentials, or otherwise create a user for the deployer

1. Upload all secrets. You'll need to override the CF cli variables to make it work nicely, since you're not deploying to a CI env
    ```
    make dev upload-all-secrets \
    DEPLOY_ENV=test-ci \
    CF_USER=user \
    CF_PASSWORD=password
    ```
1. Upload all pipelines
    ```
     make dev pipelines \
     CF_DEPLOY_ENV=andyhunt \
    DEPLOY_ENV=test-ci \
    BRANCH=upload_remaining_secrets_to_credhub_172429953
    ```
1. Run the `setup` pipeline
1. See the individual commits for testing instructions
1. Run a couple of the pipelines (e.g.`rubbernecker`, `hackmd`) to ensure they still work 

Who can review
--------------
Anyone with access to create a build env, and a dev env.
